client/src/App.js

```diff
@@ -24,54 +33,74 @@ function App() {
   }, [])
 
   return (
-    <AppWrapper>
-      <Switch>
-        <Route path="/" exact>
-          <HomePage title="Home" />
-        </Route>
-        <Route path="/createPage">
-          <CreatePage onSubmit={handleCreatePage} title="Create Page" />
-        </Route>
-        <Route path="/usersPage">
-          <UsersPage
-            users={users}
-            deleteUser={handleDeleteUser}
-            title="Users Page"
+    <ThemeProvider theme={themeMode}>
+      <GlobalStyles />
+      <AppWrapper>
+        <Switch>
+          <Route path="/" exact>
+            <HomePage title="Home" onClick={themeToggler} />
+          </Route>
+          <Route path="/createPage">
+            <CreatePage onSubmit={handleCreatePage} title="Create Page" />
+          </Route>
+          <Route path="/usersPage">
+            <UsersPage
+              users={users}
+              deleteUser={handleDeleteUser}
+              title="Users Page"
+            />
+          </Route>
+          <Route path="/searchPage">
+            <SearchPage
+              users={filteredUsers}
+              title="Search Page"
+              onChange={event => handleSearchUser(event)}
+            />
+          </Route>
+        </Switch>
+        <Route paths={['/', 'createPage', 'usersPage', 'searchPage']}>
+          <NavBar
+            pages={[
+              { title: 'Home', id: '/' },
+              { title: 'New user', id: 'createPage' },
+              { title: 'Users', id: 'usersPage' },
+              { title: 'Search', id: 'searchPage' },
+            ]}
           />
         </Route>
-        <Route path="/searchPage">
-          <SearchPage
-            users={filteredUsers}
-            title="Search Page"
-            onChange={event => handleSearchUser(event)}
-          />
-        </Route>
-      </Switch>
-      <Route paths={['/', 'createPage', 'usersPage', 'searchPage']}>
-        <NavBar
-          pages={[
-            { title: 'Home', id: '/' },
-            { title: 'New user', id: 'createPage' },
-            { title: 'Users', id: 'usersPage' },
-            { title: 'Search', id: 'searchPage' },
-          ]}
-        />
-      </Route>
-    </AppWrapper>
+      </AppWrapper>
+    </ThemeProvider>
   )
 
   function handleCreatePage(user) {
+    createNewUserEntrie(user)
+  }
+
+  function handleDeleteUser(id) {
+    const updatedUsers = users.filter(user => user._id !== id)
+    setUsers(updatedUsers)
+
+    deleteFetch(id)
+  }
+
+  function handleSearchUser(event) {
+    let value = event.target.value.toLowerCase()
+    let searchResult = []
+
+    searchResult = searchValue(value, searchResult)
```

Comment:
I don't understand the usage of `searchResult` and `searchValue`.
It's difficult to understand what kind of type is behind these names. One is an array, the other is a function (both starts with `search`.
I would consider some rules for more clear variable naming.

`value` doesn't have to be a `let`, it could be `const`.

What do you think about this? (requires further changes)

```suggestion
    const value = event.target.value.toLowerCase()
    const users = findUsers(value)
```